packages/manager/apps/pci-instances/src/pages/instances/create/components/BillingChoice.component.tsx

```diff
@@ -0,0 +1,122 @@
+import {
+  Divider,
+  Radio,
+  RadioControl,
+  RadioGroup,
+  RadioLabel,
+  Text,
+} from '@ovhcloud/ods-react';
+import { useTranslation } from 'react-i18next';
+import { PciCard } from '@/components/pciCard/PciCard.component';
+import { TInstanceCreationForm } from '../CreateInstance.page';
+import { Controller, useFormContext, useWatch } from 'react-hook-form';
+import { BILLING_TYPE } from '@/types/instance/common.type';
+import {
+  ButtonType,
+  PageLocation,
+  useOvhTracking,
+} from '@ovh-ux/manager-react-shell-client';
+
+const BillingChoice = () => {
+  const { trackClick } = useOvhTracking();
+  const { t } = useTranslation('creation');
+  const { control, setValue } = useFormContext<TInstanceCreationForm>();
+
+  const [selectedBillingType] = useWatch({
+    control,
+    name: [
+      'billingType',
+      'flavorId',
+      'distributionImageVersion.distributionImageVersionName',
+    ],
+  });
+
+  const handleSelectBilling = (billingType: BILLING_TYPE | null) => {
+    if (!billingType) return;
+    updateBillingSelection(billingType);
+  };
+
+  const updateBillingSelection = (billingType: BILLING_TYPE) => {
+    setValue('billingType', billingType);
+
+    trackClick({
+      location: PageLocation.funnel,
+      buttonType: ButtonType.button,
+      actionType: 'action',
+      actions: ['add_instance', 'select_billing', billingType],
+    });
+  };
+
+  return (
+    <section>
+      <Divider spacing="64" />
+      <div>
+        <Text preset="heading-3">
+          {t('creation:pci_instance_creation_select_billing_choice_title')}
+        </Text>
+      </div>
+      <Controller
+        name="billingType"
+        control={control}
+        render={() => (
+          <RadioGroup
+            value={selectedBillingType}
+            onValueChange={({ value }) =>
```

Comment:
Question: I don't really get why there is an update on value here AND on each individual radio. Feels like it will trigger unecessary 'setValue'. And also is it necessary to hae the value in the `<RadioGroup>` if there is already a checkwith `selected={selectedBillingType === BILLING_TYPE.Hourly}` in each `<Radio>` ?